### ISSUE TITLE
[A11y] Screen reader announces User message twice when using Directline

### ISSUE THREAD

**Issue by agriniex** [microsoft/BotFramework-WebChat#3397]
When using directline, Announcer is reading user message twice before proceeding to bot message. This behavior is noticed only when using directline. On emulator, screen reader reads user message only once then continues with bot message. 

I have a feeling that this could be related to the sequence of how activityMiddleware processes activities. Below are some list of activityId's taken from each iteration of activityMiddleware both with directline and using emulator. 

emulator:
_empty_
22527030-dae7-11ea-a51d-a5f7df81d13c -- previous bot activity
24e5c540-dae7-11ea-a51d-a5f7df81d13c -- user activity
22527030-dae7-11ea-a51d-a5f7df81d13c
24e5c540-dae7-11ea-a51d-a5f7df81d13c
22527030-dae7-11ea-a51d-a5f7df81d13c
2575f750-dae7-11ea-a51d-a5f7df81d13c -- bot response from user activity
259406a0-dae7-11ea-a51d-a5f7df81d13c 
2575f750-dae7-11ea-a51d-a5f7df81d13c
24e5c540-dae7-11ea-a51d-a5f7df81d13c

directline:
_empty_
**IvBPoaBsTYE6vz49XhrjY1-h|0000011** -- previous bot activity
IvBPoaBsTYE6vz49XhrjY1-h|0000012  -- user activity
**IvBPoaBsTYE6vz49XhrjY1-h|0000011**
IvBPoaBsTYE6vz49XhrjY1-h|0000013 -- bot response from user activity
IvBPoaBsTYE6vz49XhrjY1-h|0000014
IvBPoaBsTYE6vz49XhrjY1-h|0000013
IvBPoaBsTYE6vz49XhrjY1-h|0000012
IvBPoaBsTYE6vz49XhrjY1-h|0000012
**IvBPoaBsTYE6vz49XhrjY1-h|0000011**

I am trying to understand why there is a different sequence of which activities are processed in middleware when using directline and emulator and is there any possibility to influence this sequence? And If the fact that previous bot activity is on the last iteration could make announcer read user message twice?

**Comment by compulim** [comment]
Azure Bot Services is a distributed system and subject to network latencies and various factors. That means, activity going in and out may not be in exact order. We do sort activities using timestamp. All activities are timestamped when they arrive on the Azure Bot Services.

Emulator, in contrast, is emulating Azure Bot Services. The emulated service is locally-hosted and not distributed. Thus, activity you see here should be in an exact order as the bot send. This is because everything is local and not subject to any network latencies.

Given that, if activities are sending fast enough through Azure Bot Services. You might see activities going out-of-order.

SDK might have ways to help you queue the outgoing activities. I am not sure as I don't work on the SDK. But you can post on their GitHub repo and ask.

Are there any accessibility specific issues here?

**Comment by agriniex** [comment]
Thanks @compulim , I think what you said about "_Given that, if activities are sending fast enough through Azure Bot Services. You might see activities going out-of-order._" should be what is causing our accessibility issues.

The issue with accessibility is that this causes user message to be read twice by assistive technologies if activities are received too quickly. I was setting delays for bot messages from service side and announcer was reading messages properly.

However, purposefully slowing chat would not be desirable. Is there any way to prevent announcer from reading messages twice in this case?

**Comment by cwhitten** [comment]
@agriniex we cannot reproduce this, and neither have some large customers of web-chat using the control in a very distributed, high throughput scenario. 

Can you update your version of the control to the latest and see if the issue continues to persist?

**Comment by agriniex** [comment]
We cannot reproduce this issue as well anymore, though there has been quite a lot of changes from our side, might be that some of those changes fixed this issue. Closing this
